samples/snippets.py

```diff
@@ -0,0 +1,479 @@
+]
+
+
+# [START sqlalchemy_spanner_autocommit_on]
+def enable_autocommit_mode(url):
+    """Enables autocommit mode."""
+    conn = create_engine(url).connect()
+    level = conn.get_isolation_level()
+
+    print("Connection autocommit default mode is {}".format(level))
+    print(
+        "Spanner DBAPI default autocommit mode is {}".format(
+            conn.connection.connection.autocommit
+        )
+    )
+
+    conn.execution_options(isolation_level="AUTOCOMMIT")
+    print("Connection autocommit mode is {}".format(level))
+    print(
+        "Spanner DBAPI autocommit mode is {}".format(
+            conn.connection.connection.autocommit
+        )
+    )
+    return conn
+
+
+# [END sqlalchemy_spanner_autocommit_on]
+
+
+# [START sqlalchemy_spanner_create_table]
+def create_table(url, random_table_id):
+    """Create the table."""
+    engine = create_engine(url)
+    metadata = MetaData(bind=engine)
+
+    table = Table(
+        random_table_id,
+        metadata,
+        Column("user_id", Integer, primary_key=True),
+        Column("user_name", String(16), nullable=False),
+    )
+    table.create()
+
+    print("Table {} is created successfully".format(table.name))
+    return table
+
+
+# [END sqlalchemy_spanner_create_table]
+
+
+# [START sqlalchemy_spanner_drop_table]
+def drop_table(url, random_table_id):
+    """Drop the table."""
+    engine = create_engine(url)
+    metadata = MetaData(bind=engine)
+
+    table = Table(
+        random_table_id,
+        metadata,
+        Column("user_id", Integer, primary_key=True),
+        Column("user_name", String(16), nullable=False),
+    )
+
+    table.create()
+
+    table.drop()
+    print("Table {} is dropped successfully".format(table.name))
+    return table
+
+
+# [END sqlalchemy_spanner_drop_table]
+
+
+# [START sqlalchemy_spanner_get_table_names]
+def get_table_names(url):
+    """Retrieve the list of the table names.
+
+    The table must already exist and can be created using
+    `create_table.`
+    """
+    engine = create_engine(url)
+    insp = inspect(engine)
+    tables = insp.get_table_names()
+
+    print("Table names are:", tables)
+    return tables
+
+
+# [END sqlalchemy_spanner_get_table_names]
+
+
+# [START sqlalchemy_spanner_create_unique_indexes]
+def create_unique_indexes(table_id):
+    """Create unique index.
+
+    The table must already exist and can be created using
+    `create_table.`
+    """
+    index = Index("some_index", table_id.c.user_name, unique=True)
+    index.create()
+    print("Index created successfully")
+
+
+# [END sqlalchemy_spanner_create_unique_indexes]
+
+
+# [START sqlalchemy_spanner_delete_all_rows]
+def delete_all_rows(table):
+    """Delete all rows from the table.
+
+    The table must already exist and can be created using
+    `create_table.`
+    """
+    insert_data(table, DATA_2)
+
+    result = table.select().execute().fetchall()
+    print("Total inserted rows:", len(result))
```

Comment:
Inserting the data should be part of the test, not the sample. Also applies to the below samples (except for `insert_row` sample).